src/Builder/BuilderEncoder.php

```diff
@@ -67,7 +67,7 @@ public function canEncode(mixed $value): bool
         return (bool) $this->getEncoderFor($value)?->canEncode($value);
     }
 
-    public function encode(mixed $value): stdClass|array|string|int
+    public function encode(mixed $value): mixed
```

Comment:
Is `mixed` necessary here? I noticed it's absent on Encoder interface, which only specifies:

```
* @return mixed
* @psalm-return BSONType
```

I also noted that the original return type prior to this PR corresponds to the psalm-template implementation.

Reply:
I reverted to `Type|stdClass|array|string|int`, as this is expected.